src/ir/auth_logic/ast.h

```diff
@@ -112,61 +113,65 @@ class Fact;
 
 // CanSay corresponds to an expression of the form <principal> canSay Fact
 class CanSay {
-  public:
-    explicit CanSay(Principal principal, std::unique_ptr<Fact>& fact)
+ public:
+  explicit CanSay(Principal principal, std::unique_ptr<Fact>& fact)
       : principal_(principal), fact_(std::move(fact)) {}
-    const Principal& principal() const { return principal_; }
-    const Fact* fact() const { return fact_.get(); }
-  private:
-    Principal principal_;
-    std::unique_ptr<Fact> fact_;
+  const Principal& principal() const { return principal_; }
+  const Fact* fact() const { return fact_.get(); }
+
+ private:
+  Principal principal_;
+  std::unique_ptr<Fact> fact_;
 };
 
-// Fact corresponds to either a base fact or a an expression of the form 
-// <principal> canSay 
+// Fact corresponds to either a base fact or a an expression of the form
+// <principal> canSay <Fact>
+using FactVariantType = std::variant<BaseFact, std::unique_ptr<CanSay>>;
```

Comment:
Similar comment to the other `using` alias, unfortunately.

Reply:
see other thread.